src/components/CityCard.tsx

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from 'react';
+import { observer } from "mobx-react-lite";
+import { useStore } from "stores";
+import { CityWithUTCOffset } from 'types';
+import './CityCard.scss';
+
+function formatAMPM(date: Date) {
+    let hours = date.getHours();
+    const minutes = date.getMinutes();
+    const ampm = hours >= 12 ? 'pm' : 'am';
+    hours = hours % 12;
+    hours = hours ? hours : 12;
+    const minutesStr = minutes < 10 ? `0${minutes}` : `${minutes}`;
+    const strTime = `${hours}:${minutesStr} ${ampm}`;
+    return strTime;
+}
+
+interface CityCardProps {
+    city?: CityWithUTCOffset;
+}
+
+export const CityCard = observer(({ city }: CityCardProps) => {
```

Comment:
There is one important optimization technique:
You can rerender only <div>Time: {time}</div> on time updates, how can we achieve that?